avm/src/vm/event.rs

```diff
@@ -189,56 +186,77 @@ impl HandlerFragment {
   /// very high volume of small IO requests, but as most IO operations are several orders of
   /// magnitude slower than CPU operations, this is considered to be a very small minority of
   /// workloads and is ignored for now.
-  pub async fn run(mut self: HandlerFragment, mut hand_mem: HandlerMemory) -> HandlerMemory {
-    task::spawn(async move {
-      let mut instructions = self.get_instruction_fragment();
-      loop {
-        // io-bound fragment
-        if !instructions[0].opcode.pred_exec {
-          let mem = Arc::new(RwLock::new(hand_mem));
-          let futures: Vec<EmptyFuture> = instructions.iter().map(|ins| {
+  pub async fn run_local(mut self: HandlerFragment, mut hand_mem: HandlerMemory) -> HandlerMemory {
+    let mut instructions = self.get_instruction_fragment();
+    loop {
+      // io-bound fragment
+      if !instructions[0].opcode.pred_exec {
+        if instructions.len() > 1 {
```

Comment:
why is this necessary?

Reply:
It's not, anymore, but it was a nice shortcut when we were going to run them in parallel since it would limit the number of forks (memcpys) created when it wasn't necessary at all.

Reply:
pretty much 100% of the perf boost of this PR is just eliminating useless memcpys.